src/webgl/p5.RendererGL.js

```diff
@@ -2039,6 +2041,17 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
   _setFillUniforms(fillShader) {
     fillShader.bindShader();
 
+    if (this._useMetalness > 0 && !this.curFillColor.every((value, index) =>
+      value === 1)) {
+      const metalnessFactor = Math.min(this._useMetalness / 100, 0.6);
+      const nonMetalnessFactor = Math.max(1 - metalnessFactor, 0.4);
+
+      this.curSpecularColor = this.curSpecularColor.map(
```

Comment:
By modifying its current value, I think that means the color will be slightly different for every successive shape we render. Instead maybe we can do something like `let specularColor = [...this.curSpecularColor]` outside of the if statement and modify that, and pass that in to the uniform on line 2066 below? That way we aren't storing the new color, so it gets recalculated fresh for the next object.